ethclient/ethclient.go

```diff
@@ -147,7 +147,11 @@ func (ec *Client) TransactionByHash(ctx context.Context, hash common.Hash) (*typ
 	var tx *types.Transaction
 	err := ec.c.CallContext(ctx, &tx, "eth_getTransactionByHash", hash)
 	if err == nil {
-		if _, r, _ := tx.SignatureValues(); r == nil {
+		var signer types.Signer = types.HomesteadSigner{}
+		if tx.Protected() {
+			signer = types.NewEIP155Signer(tx.ChainId())
+		}
+		if _, r, _ := types.SignatureValues(signer, tx); r == nil {
```

Comment:
You can use RawSignatureValues here (and below). The point of this check is to see whether the fields were set in JSON.